### DISCUSSION TITLE
Theres an error somewhere in my code and I cannot find it (new coder)

### DISCUSSION THREAD

**Discussion by treeriot** [ImageMagick/ImageMagick#2348] [Help] [answered]
Hello, i'm a new coder attempting to make contact sheets of my scanned analog photographs. Below is the code i've put together, but theres an error and I don't know where. I'm on a mac.

When I run the code in terminal everything seems to run perfect, until the very last second, then it aborts and says "no such file or directory". I dropped the file right into terminal though, so I know the folder and directory is correct?

`montage -verbose -label '%f' -font Helvetica -pointsize 10 *.jpg -background '#000000' -fill 'gray' -define jpeg:size=200x200 -geometry 200x200+2+2 -auto-orient lydialunch{040..041}.JPG ~/Users/nico$/Desktop/testfolder/contact-sheet-0017.jpg`

![Screen Shot 2020-07-19 at 8 18 07 PM](https://user-images.githubusercontent.com/69062957/89095955-68850e80-d3a0-11ea-9e46-d978df3012d8.png)

Thanks so much everyone. I'm in the middle of a medical treatment right now thats boggling my brain. I've been trying to figure this out and my eyes are crossing over.
-N

**Comment by snibgo** [comment]
The names of your files seem to contain spaces: "lydialunch 040.jpg" etc. But your command does not have this space.

I suggest you remove any spaces from filenames. They just create problems.

In your command, your output directory name contains a "$", "nico$". Remove it.

**Reply by treeriot** [reply]
Thanks for the response!
I did remove the spaces in the filenames and there was no change.

Concerning the output directory name, it seems to do that only in terminal, but i never put it there. I cannot delete it from the terminal screen, nor can I see the '$' when I look at that drive in Finder.

**Comment by snibgo** [comment]
What version of IM, on what platform?

Please show the current filenames, and your command, and the text output.

From your first post, your bash does not seem to have expanded "lydialunch{040..041}.JPG" in the way I would expect. I would expect this to expand to "lydialunch040.JPG lydialunch041.JPG" with leading zeros, but your bash seems not to use leading zeros.

For now, I suggest you explicitly use the two filenames instead of one filename with "{040..041}".

**Reply by treeriot** [reply]
Thank you for the response! I included screenshots of everything below in the ways i've tried previous, as well as a shot of my imagemagick and Mac version. I'm going to try and mess with the image titles now.

**Comment by treeriot** [comment]
![Screen Shot 2020-08-01 at 11 09 28 PM](https://user-images.githubusercontent.com/69062957/89114595-4052e980-d44c-11ea-9039-253cdd018b8b.png)

**Comment by treeriot** [comment]
![Screen Shot 2020-08-01 at 11 26 28 PM](https://user-images.githubusercontent.com/69062957/89114819-8ad56580-d44e-11ea-9df4-b7b732cef162.png)

**Comment by treeriot** [comment]
No change when I adjust the filenames.. :(

![Screen Shot 2020-08-01 at 11 48 01 PM](https://user-images.githubusercontent.com/69062957/89115057-7b0b5080-d451-11ea-88b7-dde7ec450f75.png)

**Comment by snibgo** [comment]
You have done a "cd" to "desktop", not to where the input files are.

And you have "*.jpg" in the montage command. Why?

**Reply by treeriot** [reply]
Ooohh, sorry about that. 
Yes, I've tried messing with that and still get the same error.
![Screen Shot 2020-08-02 at 12 36 47 AM](https://user-images.githubusercontent.com/69062957/89115627-8dd55380-d458-11ea-87d4-95f47206e167.png)

**Comment by snibgo** [comment]
Hooray! It is reading the input files.

You have "*.jpg" in your command. Why?

You now have a _different_ error. Perhaps the directory "~/Users/nico$/" doesn't exist.

Hint: the bash command "ls" is useful for finding what files exist in the current directory, or whether some other directory exists.

**Reply by treeriot** [reply]
I believe I got the "*.jpg" code from this website, though now that i'm reading it again I believe it's redundant if I am naming files individually?
https://patdavid.net/2013/04/using-imagemagick-to-create-contact.html?fbclid=IwAR0dTr0hmmKL61frkyhaWl5r5UmRfDXK0OWQqLaWITYgN7Q5IkEpMPBNqMM

**Reply by treeriot** [reply]
I think Terminal added the "$" on its own..?? I don't know why. The path bars are visible in some of my screenshots and it just says "nico"


![Screen Shot 2020-08-02 at 1 22 13 AM](https://user-images.githubusercontent.com/69062957/89116159-e7408100-d45e-11ea-8525-ff32bdb01a0f.png)

**Reply by treeriot** [reply]
Also, bash command is completely foreign. I don't know what it means to use it.
https://programminghistorian.org/en/lessons/intro-to-bash
I'm playing with this website trying to figure it out.

here's a quick thing..

![Screen Shot 2020-08-02 at 9 51 54 PM](https://user-images.githubusercontent.com/69062957/89138739-82dafb80-d50a-11ea-8309-fbca87ad868a.png)

**Reply by fmw42** [reply]
The $ is just a prompt.

**Comment by snibgo** [comment, marked as answer]
I repeat:
> Perhaps the directory "~/Users/nico$/" doesn't exist.

I suspect you don't understand the directory structure on your computer. Tilde "\~" stands for your home directory. This seems to be "/Users/nico/". So with the tilde in front, "\~/Users/nico$/" expands to "/Users/nico/Users/nico$/". But I think that directory does not exist.

For the output file, I suggest you remove its directory entirely, for now. So the file will be written to your current directory.

**Reply by treeriot** [reply]
That worked!!!!!
thank you so much @snibgo !!!

![Screen Shot 2020-08-04 at 3 02 24 PM](https://user-images.githubusercontent.com/69062957/89333858-98176d80-d663-11ea-928d-dbf76b15dbf0.png)
